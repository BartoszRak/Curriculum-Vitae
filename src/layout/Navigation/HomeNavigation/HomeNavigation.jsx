import React from 'react'
import PropTypes from 'prop-types'
import { Toolbar } from '@material-ui/core'
import { injectIntl } from 'react-intl'

import MobileMenu from './MobileMenu'
import DesktopMenu from './DesktopMenu'

import useStyle from './HomeNavigation.style'

export function HomeNavigation({ classes: overridingClasses, intl }) {
  const classes = { ...useStyle(), ...overridingClasses }
  const { formatMessage } = intl
  const routes = [
    {
      elementId: '#commercial',
      name: formatMessage({ id: 'layout.navigation.items.commercial' }),
    },
    {
      elementId: '#projects',
      name: formatMessage({ id: 'layout.navigation.items.projects' }),
    },
    {
      elementId: '#bio',
      name: formatMessage({ id: 'layout.navigation.items.bio' }),
    },
    {
      elementId: '#skills',
      name: formatMessage({ id: 'layout.navigation.items.skills' }),
    },
    {
      elementId: '#workflow',
      name: formatMessage({ id: 'layout.navigation.items.workflow' }),
    },
    {
      elementId: '#hobbies',
      name: formatMessage({ id: 'layout.navigation.items.passions' }),
    },
  ]

  return (
    <Toolbar className={classes.toolbar}>
      <div>
        <MobileMenu className={classes.mobile} routes={routes} />
        <DesktopMenu className={classes.desktop} routes={routes} />
      </div>
    </Toolbar>
  )
}

HomeNavigation.propTypes = {
  classes: PropTypes.object,
  intl: PropTypes.object,
}

export default injectIntl(HomeNavigation)
