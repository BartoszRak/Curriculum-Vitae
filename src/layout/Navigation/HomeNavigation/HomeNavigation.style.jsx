import { makeStyles } from '@material-ui/core'

export default makeStyles(({ breakpoints }) => ({
  desktop: {
    display: 'block',
    [breakpoints.down('md')]: {
      display: 'none',
    },
  },
  mobile: {
    display: 'none',
    [breakpoints.down('md')]: {
      display: 'block',
    },
  },
  toolbar: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    width: '100%',
  },
}))
