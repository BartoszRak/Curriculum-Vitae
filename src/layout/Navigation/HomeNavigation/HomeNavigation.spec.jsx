import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { HomeNavigation } from './HomeNavigation'

jest.unmock('@material-ui/core')

jest.mock('./MobileMenu', () => 'MobileMenuMock')
jest.mock('./DesktopMenu', () => 'DesktopMenuMock')

let props

describe('component: HomeNavigation', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      intl: {
        formatMessage: jest.fn(),
      },
    }
  })
  describe('rendering', () => {
    test('render without crush', () => {
      const wrapper = shallow(<HomeNavigation {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = shallow(<HomeNavigation {...props} />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render mobile and desktop menu', () => {
      const wrapper = shallow(<HomeNavigation {...props} />)

      expect(wrapper.find('MobileMenuMock').length).toBe(1)
      expect(wrapper.find('DesktopMenuMock').length).toBe(1)
    })
  })
})
