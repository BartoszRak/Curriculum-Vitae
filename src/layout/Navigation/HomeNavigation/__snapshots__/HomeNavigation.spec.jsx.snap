// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: HomeNavigation rendering match snapshot  1`] = `
<WithStyles(ForwardRef(Toolbar))
  className="makeStyles-toolbar-3"
>
  <div>
    <MobileMenuMock
      className="makeStyles-mobile-2"
      routes={
        Array [
          Object {
            "elementId": "#commercial",
            "name": undefined,
          },
          Object {
            "elementId": "#projects",
            "name": undefined,
          },
          Object {
            "elementId": "#bio",
            "name": undefined,
          },
          Object {
            "elementId": "#skills",
            "name": undefined,
          },
          Object {
            "elementId": "#workflow",
            "name": undefined,
          },
          Object {
            "elementId": "#hobbies",
            "name": undefined,
          },
        ]
      }
    />
    <DesktopMenuMock
      className="makeStyles-desktop-1"
      routes={
        Array [
          Object {
            "elementId": "#commercial",
            "name": undefined,
          },
          Object {
            "elementId": "#projects",
            "name": undefined,
          },
          Object {
            "elementId": "#bio",
            "name": undefined,
          },
          Object {
            "elementId": "#skills",
            "name": undefined,
          },
          Object {
            "elementId": "#workflow",
            "name": undefined,
          },
          Object {
            "elementId": "#hobbies",
            "name": undefined,
          },
        ]
      }
    />
  </div>
</WithStyles(ForwardRef(Toolbar))>
`;
